lib/mix/test/mix/tasks/do_test.exs

```diff
@@ -10,4 +10,12 @@ defmodule Mix.Tasks.DoTest do
       assert_received {:mix_shell, :info, ["mix compile.app" <> _]}
     end
   end
+
+  test "run given tasks without spaces between args", context do
+    in_tmp context.test, fn ->
+      Mix.Tasks.Do.run ["compile", "--list,help"]
```

Comment:
I agree. I would rather unit test `gather_command`, but would it be ok to make it public?